File: clients/javascript/src/model-options.ts
/** Per-call settings. Omitted values use this operation's runtime environment. */
export interface ModelOptions {
  provider?: 'openai' | 'xai' | 'anthropic' | 'google' | 'openai-compatible' | 'local';
  model?: string;
  /** The AI provider's API base URL (the model endpoint, not the site under
   * test). Empty string resets the endpoint to the provider default. */
  aiBaseURL?: string;
  /** Empty string clears an inherited effort. */
  reasoningEffort?: string;
}

/** @internal Only public model settings cross the existing runtime transport. */
export function modelParams(options: ModelOptions): Record<string, unknown> {
  return Object.fromEntries(['provider', 'model', 'aiBaseURL', 'reasoningEffort']
    .filter(key => options[key as keyof ModelOptions] !== undefined)
    .map(key => [key, options[key as keyof ModelOptions]]));
}

File: clients/java/src/main/java/com/vibium/types/ModelOptions.java
package com.vibium.types;

/** Per-call model settings. Credentials are read by the runtime from its environment. */
public abstract class ModelOptions<T extends ModelOptions<T>> {
    private String provider;
    private String model;
    private String aiBaseURL;
    private String reasoningEffort;
    protected abstract T self();
    public T provider(String value) { provider = value; return self(); }
    public T model(String value) { model = value; return self(); }
    /** The AI provider's API base URL (the model endpoint, not the site under test). */
    public T aiBaseURL(String value) { aiBaseURL = value; return self(); }
    public T reasoningEffort(String value) { reasoningEffort = value; return self(); }
    public String provider() { return provider; }
    public String model() { return model; }
    public String aiBaseURL() { return aiBaseURL; }
    public String reasoningEffort() { return reasoningEffort; }
}
